### ISSUE TITLE
Plotter part 2: the figures — showable curves and actionable diagnostics

### ISSUE THREAD

**Issue by MatthewLacerda2** [MatthewLacerda2/TinyRefinementModel#179]
Part of #177. Builds on the loader and stats from part 1.

## Three images, split by audience

**`training_curve.png` — the hero.** The one you show people.
- train CE and held-out val CE against **tokens** on the x-axis (secondary axis in opt steps)
- perplexity on a right-hand axis, or a second panel — it is the number that compares to published models
- the LR schedule overlaid or beneath, drawn from `schedules.build_learning_schedule(DECAY_STEPS)` so the anneal is visible against progress
- mark the budget horizon so "how far in are we" is obvious at a glance

**`throughput_progress.png`** — tokens/sec over wall-clock, progress against `TRAIN_TOKEN_BUDGET`, ETA. Derive throughput from the supervisor log's hourly heartbeats or from step deltas; state which, and label it sampled.

**`optimization_health.png`** — the diagnostics: gradient norm, average sampled depth, zero-grad fraction, and logit health (`out_entropy`, `logz_mean`, `max_abs_logit`).

Splitting further is fine if a figure gets crowded — several readable images beat one dense one.

## The rule that matters

**Omit a panel whose data is absent.** `RunLog.has()` exists for this. The current plotter draws three flat-zero panels of reasoner-only metrics on a refiner run, which is worse than showing nothing: it looks like a measurement of zero rather than an absence.

Same for `seg1_ce`, whose meaning is arch-dependent — plot it only if it carries signal for the arch that produced the log.

## Notes from reading the current output

- The existing "Hunch-Carry Gain" shading, "Latent State Temporal Drift", "Forget Cost" and "Diversity Loss" are all reasoner concepts. The refiner has none of them (#105). Do not port them; let `has()` decide.
- The x-axis should be tokens, not steps. `TOKENS_PER_OPT_STEP` is 131,072.
- A log y-axis suits CE over a full run; it made the current curve readable.
- Dark background reads well for screenshots — keep something in that spirit, but readability first.

## Interface from part 1

```python
from instruments.runlog import load, RunLog
from instruments.model_stats import total_params, param_breakdown, vram_estimate
```

If part 1 has not landed yet, code against that interface — it is fixed.

## Constraints

**Never instantiate the model** — that is the defect being removed, and it is why the analytic path exists. No GPU. **Do not touch `trm/`**: the run is live and a crash-relaunch would execute modified trainer code. Output paths gitignored as `reasoning_analytics.png` already is. `ruff` clean.

## Verify

Run against the live run's CSV: `python -m instruments.plots --log runs/run_20260813_214725/metrics.csv`. It must also survive a short/partial CSV (a run a few steps old) and one with columns entirely blank.
